source/AI.cpp

```diff
@@ -670,23 +669,25 @@ void AI::Step(Command &activeCommands)
 	// Update the counts of how long ships have been outside the "invisible fence."
 	// If a ship ceases to exist, this also ensures that it will be removed from
 	// the fence count map after a few seconds.
-	for(auto it = fenceCount.begin(); it != fenceCount.end(); )
+	for(auto it = fenceCount.begin(); it != fenceCount.end();)
 	{
 		it->second -= FENCE_DECAY;
 		if(it->second < 0)
 			it = fenceCount.erase(it);
 		else
 			++it;
 	}
-	for(const auto &it : ships)
+	mutex lock;
+	for_each(parallel::par, ships.begin(), ships.end(), [&](const auto &it)
 	{
 		const System *system = it->GetActualSystem();
 		if(system && it->Position().Length() >= system->InvisibleFenceRadius())
 		{
+			lock_guard<mutex> guard(lock);
 			int &value = fenceCount[&*it];
 			value = min(FENCE_MAX, value + FENCE_DECAY + 1);
 		}
-	}
+	});
```

Comment:
I mean this one is so fast, doesnt really make sense to parallelize

Reply:
It's expensive because it has a square root in it, which does actually make it meaningful.

Reply:
a square root operation is not performance relevant on our scope, what is this, like 100 sqaure roots? that is nothing, these things become expensive at much higher quantities, which is not the case here